frontend/src/reducers/Reducer.ts

```diff
@@ -0,0 +1,119 @@
+import { globalStates, headerStates, bottomStates, pageStates } from "../constants/States";
+import { actionTypes } from '../constants/ActionTypes';
+import 
+{   State, 
+    Action, 
+    searchPayload, 
+    addItemPayload, 
+    removeItemPayload, 
+    updateItemsPayload, 
+    switchStorePayload
+} from "../constants/Interfaces";
+
+const initialState : State = {
+    headerState: headerStates.title,
+    bottomState: bottomStates.itemSelection,
+    pageState: pageStates.itemList,
+    store: {
+        searchQuery: '',
+        items: []
+    }
+};
+
+export default function setState(state : State = initialState, action: Action) {
```

Comment:
This function is kinda big right now. I think it would be improved if you used `combineReducers`, which is explained [here](https://redux.js.org/api/combinereducers)